microcloud/service/lxd.go

```diff
@@ -478,9 +449,93 @@ func (s LXDService) GetUplinkInterfaces(bootstrap bool, peers map[string]mdns.Se
 	return candidates, nil
 }
 
+func (s LXDService) SetupNetwork(uplinkNetworks map[string]string, networkConfig map[string]string) error {
+	client, err := s.client("")
+	if err != nil {
+		return err
+	}
+
+	if uplinkNetworks[s.Name()] == "" {
+		// Setup networking.
+		underlay, _, err := defaultGatewaySubnetV4()
+		if err != nil {
+			return fmt.Errorf("Couldn't determine Fan overlay subnet: %w", err)
+		}
+
+		underlaySize, _ := underlay.Mask.Size()
+		if underlaySize != 16 && underlaySize != 24 {
+			// Override to /16 as that will almost always lead to working Fan network.
+			underlay.Mask = net.CIDRMask(16, 32)
+			underlay.IP = underlay.IP.Mask(underlay.Mask)
+		}
+
+		network := api.NetworksPost{
+			NetworkPut: api.NetworkPut{
+				Config: map[string]string{
+					"bridge.mode":         "fan",
+					"fan.underlay_subnet": underlay.String(),
+				},
+				Description: "Default Ubuntu fan powered bridge",
+			},
+			Name: "lxdfan0",
+			Type: "bridge",
+		}
+
+		err = client.CreateNetwork(network)
+		if err != nil {
+			return err
+		}
+	} else {
+		err = client.UseTarget(s.Name()).CreateNetwork(api.NetworksPost{
+			NetworkPut: api.NetworkPut{Config: map[string]string{"parent": uplinkNetworks[s.Name()]}},
+			Name:       "UPLINK",
+			Type:       "physical",
+		})
+		if err != nil {
+			return err
+		}
+
+		network := api.NetworkPut{Config: map[string]string{}}
+		for gateway, ipRange := range networkConfig {
+			ip, _, err := net.ParseCIDR(gateway)
+			if err != nil {
+				return err
+			}
+
+			if ip.To4() != nil {
+				network.Config["ipv4.gateway"] = gateway
+				network.Config["ipv4.ovn.ranges"] = ipRange
+			} else {
+				network.Config["ipv6.gateway"] = gateway
+				network.Config["ipv6.ovn.ranges"] = ipRange
+			}
+		}
+
+		err = client.CreateNetwork(api.NetworksPost{
+			NetworkPut: network,
+			Name:       "UPLINK",
```

Comment:
I'd just add a line with `Comment: "Uplink for OVN networks"` just so we have a description for it.